lib/util/check-unsupported-builtins.js

```diff
@@ -92,11 +97,14 @@ module.exports.checkUnsupportedBuiltins = function checkUnsupportedBuiltins(
 
     for (const { node, path, info } of references) {
         const name = unprefixNodeColon(path.join("."))
-        const supported = isSupported(info, options.version)
-
-        if (supported === true || options.ignores.has(name)) {
+        if (
+            isInRange(info.supported, options.version) ||
+            options.ignores.has(name) ||
+            (options.allowExperimental === true && info.experimental != null)
```

Comment:
I wonder if we should check the current range against the experimental range here so we don't allow experimental features that are not in the configured node version